pandas/core/groupby/ops.py

```diff
@@ -624,71 +625,60 @@ def _get_splitter(self, data: NDFrame) -> DataSplitter:
         -------
         Generator yielding subsetted objects
         """
-        ids, _, ngroups = self.group_info
+        ids, ngroups = self.group_info
         return _get_splitter(
             data,
             ids,
             ngroups,
             sorted_ids=self._sorted_ids,
-            sort_idx=self._sort_idx,
+            sort_idx=self.result_ilocs,
         )
 
-    @final
-    @cache_readonly
-    def group_keys_seq(self):
-        if len(self.groupings) == 1:
-            return self.levels[0]
-        else:
-            ids, _, ngroups = self.group_info
-
-            # provide "flattened" iterator for multi-group setting
-            return get_flattened_list(ids, ngroups, self.levels, self.codes)
-
     @cache_readonly
     def indices(self) -> dict[Hashable, npt.NDArray[np.intp]]:
         """dict {group name -> group indices}"""
         if len(self.groupings) == 1 and isinstance(self.result_index, CategoricalIndex):
             # This shows unused categories in indices GH#38642
             return self.groupings[0].indices
         codes_list = [ping.codes for ping in self.groupings]
-        keys = [ping._group_index for ping in self.groupings]
-        return get_indexer_dict(codes_list, keys)
+        return get_indexer_dict(codes_list, self.levels)
 
     @final
+    @cache_readonly
```

Comment:
Does this need to be cached?

Reply:
It's being used in more places here, primarily where `_sort_idx` was used. `_sort_idx` was cached, so this still neutral on memory use.